source/reference/non_json_data.rst

```diff
@@ -0,0 +1,87 @@
+.. index::
+   single: non-JSON data
+   single: media
+
+.. _media:
+
+Media: string-encoding non-JSON data
+------------------------------------
+
+|draft7|
+
+JSON schema has a set of keywords to describe and optionally validate non-JSON
+data stored inside JSON strings. Since it would be difficult to write validators
+for many media types, JSON schema validators are not required to validate the
+contents of JSON strings based on these keywords. At the time of this writing,
+it appears that most of them do not. However, these keywords are still useful
+for an application that consumes validated JSON.
+
+.. index::
+   single: contentMediaType
+   single: media; contentMediaType
+
+contentMediaType
+````````````````
+
+The ``contentMediaType`` keyword specifies the MIME type of the contents of a
+string, as described in `RFC 2046 <https://tools.ietf.org/html/rfc2046>`_.
+There is a list of `MIME types officially registered by the IANA
+<http://www.iana.org/assignments/media-types/media-types.xhtml>`_, but the set
+of types supported will be application and operating system dependent. Mozilla
+Developer Network also maintains a `shorter list of MIME types that are
+important for the web
+<https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types>`_
+
+.. index::
+   single: contentEncoding
+   single: media; contentEncoding
+
+contentEncoding
+```````````````
+
+The ``contentEncoding`` keyword specifies the encoding used to store the
+contents, as specified in `RFC 2054, part 6.1
+<https://tools.ietf.org/html/rfc2045>`_.
+
+The acceptable values are ``7bit``, ``8bit``, ``binary``, ``quoted-printable``
+and ``base64``. If not specified, the encoding is the same as the containing
+JSON document.
+
+Without getting into the low-level details of each of these encodings, there are
+really only two options useful for modern usage:
+
+- If the string contains a binary MIME type (``image/png``, for example), set
+  ``contentEncoding`` to ``base64`` and encode the contents using `Base64
```

Comment:
I think since it's not in draft 7, I'll defer mentioning it to later updates.  Since many users I've come across are stuck on earlier drafts for various "reasons", I've tried to be really specific about when certain features are and aren't available.